### PR TITLE
Refactor updateProfileField

### PR DESCRIPTION
**Pull request by cvalarida**
I was looking through the login code today and figured I could take a quick shot at refactoring this so we didn't have to dispatch a thousand actions when logging in.

### CODE REVIEW

File: src/js/auth/containers/AuthApp.jsx [hunk 1]
```diff
@@ -112,8 +112,9 @@ const mapDispatchToProps = (dispatch) => {
     onUpdateLoggedInStatus: (update) => {
       dispatch(updateLoggedInStatus(update));
     },
+    // Is this ever used?
     onUpdateProfile: (field, update) => {
```
Comment (cvalarida) [inline src/js/auth/containers/AuthApp.jsx]: I guess I didn't really need to keep the comment, but...is this ever used? It's the only instance of `onUpdateProfile` in the file and it's not being exported or anything, so...

Still, it seemed a bit out of scope to nix it in this PR.

File: src/js/user-profile/reducers/profile.js [hunk 2]
```diff
@@ -35,8 +35,8 @@ const initialState = {
 
 function profileInformation(state = initialState, action) {
   switch (action.type) {
-    case UPDATE_PROFILE_FIELD: {
-      return _.set(action.propertyPath, action.value, state);
+    case UPDATE_PROFILE_FIELDS: {
+      return _.merge(state, action.newState);
```
Comment (jbalboni) [inline src/js/user-profile/reducers/profile.js]: I'm not sure merge is correct, I think you want to do `assign`. You could get old and new array items mixed together with merge, I think.

File: src/js/auth/containers/AuthApp.jsx (hunk 1 above)
Comment (jbalboni) [inline src/js/auth/containers/AuthApp.jsx]: Yeah, it doesn't look used to me.

**Review by jbalboni — APPROVED** [review]
LGTM, but I think we should have @annekainicUSDS or @raquelromano-gov look at it, too.

File: src/js/user-profile/reducers/profile.js (hunk 2 above)
Comment (cvalarida) [inline src/js/user-profile/reducers/profile.js]: Ohh, yeah. Good point. 👍

**Review by raquelromano-gov — APPROVED** [review]
LGTM--thanks!

**Pull request merged**